nav2_bt_navigator/src/navigators/navigate_to_pose.cpp

```diff
@@ -209,18 +209,29 @@ NavigateToPoseNavigator::initializeGoalPose(ActionT::Goal::ConstSharedPtr goal)
     feedback_utils_.global_frame, feedback_utils_.robot_frame,
     feedback_utils_.transform_tolerance);
 
+  geometry_msgs::msg::PoseStamped goal_pose;
+  if (!nav2_util::transformPoseInTargetFrame(
+        goal->pose, goal_pose, *feedback_utils_.tf, feedback_utils_.global_frame,
+        feedback_utils_.transform_tolerance)) {
+    RCLCPP_ERROR(
+      logger_,
+      "Failed to transform a goal pose provided with frame_id '%s' to the global frame '%s'.",
+      goal->pose.header.frame_id.c_str(), feedback_utils_.global_frame.c_str());
+    throw std::runtime_error("transformation failed");
```

Comment:
What catches this throw? Also make the message more verbose

Reply:
What is the expected behavior then if the goal pose (or one of the goal poses) cannot be transformed? Is it okay to change the return type of `NavigateToPoseNavigator::initializeGoalPose()` and `NavigateThroughPosesNavigator::initializeGoalPoses()` to `bool`, such that the error will be propagated to the respective `goalReceived()` callback? I assume that returning `false` here implies that the goal is rejected?

What additional information would you expect to be logged? We could add something like "goal rejected..."?

Also note that `nav2_util::transformPoseInTargetFrame()` already logs the failure [here](https://github.com/ros-planning/navigation2/blob/19a8165a75c9479e9d1bc017eab87255dc09c579/nav2_util/src/robot_utils.cpp#L50-L75) depending on the exact type of `tf2::TransformException`, so this log statement is already kind of redundant.

I also noticed that the return value of `getCurrentPose()` is currently not checked in the `onLoop()` callbacks and in `NavigateToPoseNavigator::initializeGoalPose()`. That could also fail, for example if global localization was lost and transforms timed out. Then the `current_pose` is left default initialized, with undefined behavior?

Reply:
> I assume that returning false here implies that the goal is rejected?

Correct, I think that's a good way to do it. I think your logs look good to me!

In `onPreempt`, you can also use the new return type of `initializeGoalPose` to `bt_action_server_->terminatePendingGoal();` as you see in the else statement next to it. 

In `onLoop`, you're right. Things are a little fragile there. These are primarily high level "FYI" metrics so I'm not overly concerned about them, but simple:

```
if (!  nav2_util::getCurrentPose(
    current_pose, *feedback_utils_.tf,
    feedback_utils_.global_frame, feedback_utils_.robot_frame,
    feedback_utils_.transform_tolerance))
{
  // LOG SOMETHING
  // return;
}
```
and not publish feedback - that sounds very reasonable